index.d.ts

```diff
@@ -0,0 +1,46 @@
+export type CurriedHandler<P = any, S = any> = (payload: P) => (state: S) => S;
+
+// TODO: figure out how import this from redux, since this pkg has no dependencies
+type Reducer = (state: any, action: {type: any, payload: any}) => any;
+type ActionCreator<P = any> = (payload: P) => {type: string, payload: P};
+type StrMap<T> = Record<string, T>;
+
+/** Utility to get the payload type of a CurriedHandler */
+export type HandlerPayloadType<T extends CurriedHandler> = T extends CurriedHandler<infer P> ? P : any
+
+export declare function createType(namespace: string, actionName: string): string;
+
+export declare function compileActionTypes<T extends StrMap<CurriedHandler>>(
+    namespace: string,
+    actions: T,
+): Record<keyof T, string>;
+
+export declare function compileActionCreators<T extends StrMap<string>>(types: T):
+Record<keyof T, ActionCreator>;
```

Comment:
This can just be `StrMap<ActionCreator>`, right? Or does the `keyof T` allow TypeScript to track the actual keys of the StrMap?